src/order/ideal.lean

```diff
@@ -135,8 +169,36 @@ begin
   { exact λ _, I.mem_of_le le_top topmem }
 end
 
-lemma proper_of_ne_top {I : ideal P} (ntop : I ≠ ⊤) : proper I :=
-proper_of_not_mem (λ h, ntop (top_of_mem_top h))
+lemma is_proper_of_ne_top {I : ideal P} (ne_top : I ≠ ⊤) : is_proper I :=
+is_proper_of_not_mem (λ h, ne_top (top_of_mem_top h))
+
+lemma is_proper.ne_top {I : ideal P} (hI : is_proper I) : I ≠ ⊤ :=
+begin
+  intro h,
+  rw [ext'_iff, top_coe] at h,
+  apply hI.ne_univ,
+  assumption,
+end
+
+lemma is_coatom.is_proper {I : ideal P} (hI : is_coatom I) : is_proper I :=
+is_proper_of_ne_top hI.1
+
+lemma is_proper_iff_ne_top {I : ideal P} : is_proper I ↔ I ≠ ⊤ :=
+⟨λ h, h.ne_top, λ h, is_proper_of_ne_top h⟩
+
+lemma is_maximal.is_coatom {I : ideal P} (h : is_maximal I) : is_coatom I :=
+⟨is_maximal.to_is_proper.ne_top, 
+ λ J _, by { rw [ext'_iff, top_coe], exact is_maximal.maximal_proper J ‹_› }⟩
+
+lemma is_maximal.is_coatom' {I : ideal P} [is_maximal I] : is_coatom I :=
+is_maximal.is_coatom ‹_›
+
+lemma is_coatom.is_maximal {I : ideal P} (hI : is_coatom I) : is_maximal I :=
+{ maximal_proper := λ _ _, by simp [hI.2 _ ‹_›],
+  ..is_coatom.is_proper ‹_› }
+
+lemma is_maximal_iff_is_coatom {I : ideal P} : is_maximal I ↔ is_coatom I :=
+⟨λ h, h.is_coatom, λ h, h.is_maximal⟩
```

Comment:
Ah, this is breaking because `is_coatom` is in the root namespace and not under `order.ideal` and the full name of `is_coatom.is_maximal` is `order.ideal.is_coatom.is_maximal`. After #6691 is merged we'll be able to put `is_coatom.is_maximal` in the root namespace by writing `lemma _root_.is_coatom.is_maximal`, but until then we'll have to end the sections / namespaces and then re-open them afterwards.

Let me see if we can get #6691 merged relatively quickly.

Reply:
OK #6691 is merged now; try merging `master` and using the new `_root_` syntax for the two new `is_coatom` lemmas. If it works, feel free to put on the queue:
bors d+